specs/jsonschema-validation.md

```diff
@@ -532,19 +532,24 @@ defined by [RFC 2046](#rfc2046).
 
 ### `contentSchema`
 
-If the instance is a string, and if `contentMediaType` is present, this property
-contains a schema which describes the structure of the string.
+If the instance is a string, and if `contentMediaType` is present, this
+property's subschema describes the structure of the string.
 
 This keyword MAY be used with any media type that can be mapped into JSON
 Schema's data model. Specifying such mappings is outside of the scope of this
 specification.
 
-The value of this property MUST be a valid JSON schema. It SHOULD be ignored if
-`contentMediaType` is not present. Accessing the schema through the schema
-location IRI included as part of the annotation will ensure that it is correctly
-processed as a subschema. Using the extracted annotation value directly is only
-safe if the schema is an embedded resource with both `$schema` and an
-absolute IRI `$id`.
+The value of this property MUST be a valid JSON schema. The subschema is
+produced as an annotation.
+
+Since `contentMediaType` is required to provide instruction on how to interpret
+string content, the annotation schema produced by this keyword has no meaning if
+`contentMediaType` is not present.
+
+Accessing the schema through the schema location IRI included as part of the
+annotation will ensure that it is correctly processed as a subschema. Using the
+extracted annotation value directly is only safe if the subschema is an embedded
```

Comment:
Maybe we should make this a SHOULD?

Reply:
I'm not sure what you're suggesting. This is informative, not a requirement.

Reply:
So from what I'm reading here, there are edge cases in `contentSchema` if the schema doesn't have `$id` and `$schema`. If that's the case, shouldn't we highlight in more of a "SHOULD" manner, which from what I understand, it something you should do rather than a MUST (something you HAVE to do)? Or maybe "RECOMMENDED" is the right one similar to this case: https://json-schema.org/draft/2020-12/json-schema-validation#section-7.2.2-5?

Reply:
It's not an edge case.  It just means that if you intend to use the `contentSchema` subschema solely in its own context, as you would if you received it as an annotation, then any relative `$ref`s will only be resolvable if the subschema has both `$schema` and `$id`.

This goes back to my comment [here](https://github.com/json-schema-org/json-schema-spec/issues/1381#issuecomment-2433652911) where I show a `contentSchema` subschema attempting to reference a definition in its parent schema.  If you extract the subschema (again, because you've received it as an annotation), then that reference fails.

There's not a best practice here.  Both approaches have valid use cases, and schema authors are free to do what makes sense for them.  This is merely a caution to schema authors to understand the implications of their approach.

Reply:
It's obvious to me from @karenetheridge's and your comments that this paragraph isn't clear, so I'll reword it.